cmd/finch/devcontainer_patch.go

```diff
@@ -0,0 +1,100 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package main denotes the entry point of finch CLI.
+// TODO: Remove all instances of these calls once supported upstream
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/runfinch/finch/pkg/command"
+)
+
+// nerdctl outputs multiple id inspect call as multiple arrays of json which is not parsable by json.
+//
+//	Helper function helps to separate out into individual json array and a single json is reconstructed.
+func extractOuterArrays(input string) []string {
+	var arrays []string
+	var bracketCount, start int
+
+	for i, char := range input {
+		switch char {
+		case '[':
+			if bracketCount == 0 {
+				start = i
+			}
+			bracketCount++
+		case ']':
+			bracketCount--
+			if bracketCount == 0 {
+				arrays = append(arrays, input[start:i+1])
+			}
+		}
+	}
+
+	return arrays
+}
+
+func prettyPrintJSON(input string) {
+	jsonArrays := extractOuterArrays(input)
```

Comment:
Why can't we import the type from nerdctl and then unmarshall it?

Reply:
the output from nerdctl is of the form
```
[
container1 inspect
]
[

container 2 inspect
}
]
```

this is not a valid json, so first we tract the comma and convert it to a single valid json array that docker outputs, 

Post that i think it would be feasible so that the type cast can be avoided but the conversion has to be made prior.

Reply:
What if we append `--format '{{json .}}'` to the `container inspect` command?